packages/integration-objects/test/ValidateNonPrimaryFieldsForRates.test.js

```diff
@@ -0,0 +1,148 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+const {
+  validateNonPrimaryKeyFieldsForCurrencyExchangeRate
+} = require('../srv/validations/ValidateNonPrimaryFieldsForRates');
+const ErrorStatuses = require('../srv/utils/ErrorStatuses');
+const RateExtensionConstants = require('../srv/utils/RateExtensionConstants');
+
+describe('validate valid fields among non-primary keys', function () {
+  it('does not throw any error', function () {
+    const reqObj = {
+      tenantID: 'tenant1',
+      dataProviderCode: 'TW',
+      dataSource: 'THR',
+      exchangeRateType: 'aA',
+      fromCurrencyThreeLetterISOCode: 'CND',
+      toCurrencyThreeLetterISOCode: 'INR',
+      validFromDateTime: '2020-02-28T06:38:29Z',
+      exchangeRateValue: '80.00'
+    };
+    expect(validateNonPrimaryKeyFieldsForCurrencyExchangeRate(reqObj)).toBeUndefined();
+  });
+
+  it('toCurrencyFactor negative check', () => {
+    const reqObj = {
+      tenantID: 'tenant1',
+      dataProviderCode: 'TW',
+      dataSource: 'THR',
+      exchangeRateType: 'aA',
+      fromCurrencyThreeLetterISOCode: 'INR',
+      toCurrencyThreeLetterISOCode: 'INR',
+      validFromDateTime: '2020-02-28T06:38:29Z',
+      exchangeRateValue: '80.00',
+      toCurrencyFactor: '-2'
+    };
+    try {
+      validateNonPrimaryKeyFieldsForCurrencyExchangeRate(reqObj);
+    } catch (err) {
+      expect(err.message).toBe(RateExtensionConstants.INVALID_TO_CURRENCY_FACTOR_VALUE_FIELD);
```

Comment:
same here do not use the try catch blocks in tests because jest has nice exception tests. Note that it is good to make also a negative test once i.e. breaking the assertion and see if the test really fails. Especially for async assertion this is always worth a try because sometimes one does not really reach the promise rejection one wants to test.

Reply:
This is also fixed.